helix-term/src/ui/prompt.rs

```diff
@@ -456,30 +465,28 @@ impl Prompt {
         // render buffer text
         surface.set_string(area.x, area.y + line, &self.prompt, prompt_color);
 
-        let (input, is_suggestion): (Cow<str>, bool) = if self.line.is_empty() {
-            // latest value in the register list
-            match self
+        let line_area = area.clip_left(self.prompt.len() as u16).clip_top(line);
+        if self.line.is_empty() {
+            // Show the most recently entered value as a suggestion.
+            if let Some(suggestion) = self
                 .history_register
                 .and_then(|reg| cx.editor.registers.last(reg))
-                .map(|entry| entry.into())
             {
-                Some(value) => (value, true),
-                None => (Cow::from(""), false),
+                surface.set_string(line_area.x, line_area.y, suggestion, suggestion_color);
             }
+        } else if let Some((language, loader)) = self.language.as_ref() {
+            let mut text: ui::text::Text = crate::ui::markdown::highlighted_code_block(
+                self.line.clone(),
```

Comment:
I think it should be possible to pass a `&str` instead of a `String` here (would require some additional changes but the pref concern is pretty similar to the Rope -> RopeSlice case. Its not a huge deal ofcourse but from a quick glimpse just changing the argument type should be enough